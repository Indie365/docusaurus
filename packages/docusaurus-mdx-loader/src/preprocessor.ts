/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  escapeMarkdownHeadingIds,
  unwrapMdxCodeBlocks,
  admonitionTitleToDirectiveLabel,
} from '@docusaurus/utils';
import {normalizeAdmonitionOptions} from './remark/admonitions';
import type {Options} from './loader';

export default function preprocessContent(
  md: string,
  {
    markdownConfig,
    admonitions,
  }: {
    markdownConfig: Options['markdownConfig'];
    admonitions: Options['admonitions'] | undefined;
  },
): string {
  md = unwrapMdxCodeBlocks(md);
  if (markdownConfig.mdx1Compat.headingIds) {
    md = escapeMarkdownHeadingIds(md);
  }
  if (markdownConfig.mdx1Compat.admonitions && admonitions) {
    const {keywords} = normalizeAdmonitionOptions(admonitions);
    md = admonitionTitleToDirectiveLabel(md, keywords);
  }
  return md;
}
