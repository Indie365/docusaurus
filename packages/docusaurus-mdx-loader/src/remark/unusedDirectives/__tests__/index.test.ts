/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import remark2rehype from 'remark-rehype';
import stringify from 'rehype-stringify';
import vfile from 'to-vfile';
import preprocessor from '../../../preprocessor';
import plugin from '../index';

const processFixture = async (name: string) => {
  const {remark} = await import('remark');
  const {default: directives} = await import('remark-directive');

  const filePath = path.join(__dirname, '__fixtures__', `${name}.md`);
  const file = await vfile.read(filePath);
  const fileContentPreprocessed = preprocessor({
    fileContent: file.toString(),
    filePath,
    markdownConfig: {
      mermaid: false,
      mdx1Compat: {
        admonitions: false,
        comments: false,
        headingIds: false,
      },
    },
  });

  const result = await remark()
    .use(directives)
    .use(plugin)
    .use(remark2rehype)
    .use(stringify)
    .process(fileContentPreprocessed);

  return result.value;
};

describe('directives remark plugin', () => {
  it('default behavior for custom keyword', async () => {
    const result = await processFixture('directives');
    expect(result).toMatchSnapshot();
  });
});
