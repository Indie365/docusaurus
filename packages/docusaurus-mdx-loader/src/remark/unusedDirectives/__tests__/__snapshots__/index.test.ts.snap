// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directives remark plugin - client compiler default behavior for container directives 1`] = `
"<admonition type="danger"><p>Take care of snowstorms...</p></admonition>
<div><p>unused directive content</p></div>
<p>:::NotAContainerDirective with a phrase after</p>
<p>:::</p>
<p>Phrase before :::NotAContainerDirective</p>
<p>:::</p>"
`;

exports[`directives remark plugin - client compiler default behavior for container directives 2`] = `
[
  [
    "[WARNING] We found a potential error in your documentation unusedDirective "packages/docusaurus-mdx-loader/src/remark/unusedDirectives/__tests__/__fixtures__/containerDirectives.md:7:1"",
  ],
]
`;

exports[`directives remark plugin - client compiler default behavior for leaf directives 1`] = `
"<div></div>
<p>Leaf directive in a phrase ::NotALeafDirective</p>
<p>::NotALeafDirective with a phrase after</p>"
`;

exports[`directives remark plugin - client compiler default behavior for leaf directives 2`] = `
[
  [
    "[WARNING] We found a potential error in your documentation unusedLeafDirective "packages/docusaurus-mdx-loader/src/remark/unusedDirectives/__tests__/__fixtures__/leafDirectives.md:1:1"",
  ],
]
`;

exports[`directives remark plugin - client compiler default behavior for text directives 1`] = `
"<p>Simple: textDirective1</p>
<pre><code class="language-sh">Simple: textDirective1
</code></pre>
<p>Simple<div></div></p>
<pre><code class="language-sh">Simple:textDirective1
</code></pre>"
`;

exports[`directives remark plugin - client compiler default behavior for text directives 2`] = `
[
  [
    "[WARNING] We found a potential error in your documentation textDirective2 "packages/docusaurus-mdx-loader/src/remark/unusedDirectives/__tests__/__fixtures__/textDirectives.md:7:7"",
  ],
]
`;

exports[`directives remark plugin - server compiler default behavior for container directives 1`] = `
"<admonition type="danger"><p>Take care of snowstorms...</p></admonition>
<div><p>unused directive content</p></div>
<p>:::NotAContainerDirective with a phrase after</p>
<p>:::</p>
<p>Phrase before :::NotAContainerDirective</p>
<p>:::</p>"
`;

exports[`directives remark plugin - server compiler default behavior for leaf directives 1`] = `
"<div></div>
<p>Leaf directive in a phrase ::NotALeafDirective</p>
<p>::NotALeafDirective with a phrase after</p>"
`;

exports[`directives remark plugin - server compiler default behavior for text directives 1`] = `
"<p>Simple: textDirective1</p>
<pre><code class="language-sh">Simple: textDirective1
</code></pre>
<p>Simple<div></div></p>
<pre><code class="language-sh">Simple:textDirective1
</code></pre>"
`;
